pandas_gbq/gbq.py

```diff
@@ -42,19 +41,50 @@ def _test_google_api_imports():
 
     try:
         import httplib2  # noqa
-        try:
-            from googleapiclient.discovery import build  # noqa
-            from googleapiclient.errors import HttpError  # noqa
-        except:
-            from apiclient.discovery import build  # noqa
-            from apiclient.errors import HttpError  # noqa
-        from oauth2client.client import AccessTokenRefreshError  # noqa
-        from oauth2client.client import OAuth2WebServerFlow  # noqa
-        from oauth2client.file import Storage  # noqa
-        from oauth2client.tools import run_flow, argparser  # noqa
-    except ImportError as e:
+        from google_auth_oauthlib.flow import InstalledAppFlow  # noqa
+        import google_auth_httplib2  # noqa
+    except ImportError as ex:
         raise ImportError("Missing module required for Google BigQuery "
-                          "support: {0}".format(str(e)))
+                          "support: {0}".format(str(ex)))
+
+    try:
+        from googleapiclient.discovery import build  # noqa
+        from googleapiclient.errors import HttpError  # noqa
+    except ImportError as ex:
+        raise ImportError(
+            "pandas requires google-api-python-client for Google  BigQuery "
+            "support: {0}".format(str(ex)))
+
+    try:
+        import google.auth  # noqa
+    except ImportError as ex:
+        raise ImportError(
+            "pandas requires google-auth for Google  BigQuery support: "
+            "{0}".format(str(ex)))
+
+    _check_google_client_version()
+
+
+def _try_credentials(project_id, credentials):
+    import httplib2
+    from googleapiclient.discovery import build
+    import googleapiclient.errors
+    from google_auth_httplib2 import AuthorizedHttp
+
+    if credentials is None:
+        return None
+
+    http = httplib2.Http()
+    try:
+        http = AuthorizedHttp(credentials, http=http)
```

Comment:
nit: shadowing variables makes me uncomfortable, I'd prefer you call this one `authed_http`.